app/helpers/dams_objects_helper.rb

```diff
@@ -788,7 +788,7 @@ def grabRestrictedText(data)
   def grab_access_text(document)
     out = []
     data = rights_data document
-    return nil unless metadata_display?(data) && cannot?(:read, document)
+    return nil unless metadata_display?(data) && (current_user.nil? || local_user_public_col?(@document))
```

Comment:
@lsitu - I already tried different ability check including cannot?(:read, document) for all type of users and they all return false.  I couldn't use it to check for local or public user access local or public collection.

Reply:
@lsitu - here are my test results to print out that ability check
local user - public collection : cannot? read - returns false
local user - local collection : cannot? read - returns false
public user - local collection : cannot? read - returns false
I need a check for local user to access public collection with metadatadisplay object.

Reply:
@lsitu - You can try that condition for this test https://github.com/ucsdlib/damspas/pull/480/files#diff-bff6a4c6341ffbdd45f5cabab7685905R1037  and https://github.com/ucsdlib/damspas/pull/480/files#diff-bff6a4c6341ffbdd45f5cabab7685905R1121  Thanks

Reply:
@VivianChu But either rspec or manual tests seems to be correct for localDisplay objects. So I am wondering whether there is an issue with setting up the test as what I'd seen before. Let me run your tests to see how it goes.

Reply:
@VivianChu I see the problem. Your test at https://github.com/ucsdlib/damspas/pull/480/files#diff-bff6a4c6341ffbdd45f5cabab7685905R1037 is failed with ability check `cannot?(:read, document)` because the object you setup for the test is a public object (`copyright_attributes: [{status: 'Public domain'}]`: https://github.com/ucsdlib/damspas/pull/480/files#diff-bff6a4c6341ffbdd45f5cabab7685905R997). With the rules we'd setup in the past for access control, this is always a public object with no access control overriding. I don't know whether we'll have such a case in the future or not, but we may need a ticket to fix it at the access control level if this will become a real case. For now, I think you can change it to something like `copyright_attributes: [{status: 'Unknown'}]`. In this case, access override will take it's role to override it for metadataDisplay and your tests for public metadata-only accesses will pass.

Reply:
@lsitu - I'll update it to use other copyright_attributes "Under copyright" and change it to use cannot? read. Thanks